gym_solo/core/termination.py

```diff
@@ -0,0 +1,48 @@
+from abc import ABC, abstractmethod
+
+
+class Termination(ABC):
+    @abstractmethod
+    def __init__(self):
+        """This is used to define the termination condition
+        """
+        pass
```

Comment:
Don't need to make the init an abstract method unless you're trying to force additional arguments to the constructor.

Reply:
My idea was to support if the actual classes wanted to have additional arguments. Kind of similar to what we had in observations

Reply:
Ahh I see. In the observations, we forced that the `body_id` is part of every child of the class. However, the child class can do something like:

```python
def __init__(self, body_id, arg1, arg2):
  pass
```

Notice that `arg1` and `arg2` aren't dependent upon the parent. Basically, if you want to force that *all* of the children have a common argument, you would make an abstract `__init__` method. Otherwise, you don't really need to add it cause python does it for you.

Reply:
Adding body_id as an argument because any reference to pybullet to calculate the termination condition for a robot will require that